storage/storage_reference.go

```diff
@@ -30,6 +33,8 @@ func newReference(transport storageTransport, reference, id string, name referen
 		reference: reference,
 		id:        id,
 		name:      name,
+		tag:       tag,
+		digest:    digest,
```

Comment:
(Also `reference` vs. `name`+`tag+`digest` seem redundant at a first glance, but I haven’t checked in detail.)